contracts/core/Registry.sol

```diff
@@ -1,242 +1,110 @@
-// SPDX-License-Identifier: AGPL-3.0-only
-pragma solidity 0.8.19;
-
-import {Metadata} from "../core/libraries/Metadata.sol";
-import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
-
-contract Registry is Initializable {
-    // A linked list of owners of a identity
-    // The use of a linked list allows us to easily add and remove owners,
-    // access them directly in O(1), and loop through them.
-    //
-    // {
-    //     count: 3,
-    //     list: {
-    //         OWNERS_LIST_SENTINEL => owner1Address,
-    //         owner1Address => owner2Address,
-    //         owner2Address => owner3Address,
-    //         owner3Address => OWNERS_LIST_SENTINEL
-    //     }
-    // }
-    struct OwnerList {
-        uint256 count;
-        mapping(uint => address) list;
-    }
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { OwnableRoles } from "../../lib/solady/src/auth/OwnableRoles.sol";
+import { Metadata } from "./libraries/Metadata.sol";
+
+// use Solady Roles for ownership of projects: https://github.com/Vectorized/solady/blob/main/src/auth/OwnableRoles.sol
+contract Registry is OwnableRoles {
+
+    uint private identityId;
 
     struct IdentityDetails {
-        uint id;
         string name;
-        Metadata.MetaPtr metadata;
-        address attestationAddress;
+        Metadata metadata;
+        address attestationAddr;
+        // address coreOwner; // ASK: Should this be added?
     }
 
-    // State variables
-
-    // Used as sentinel value in the owners linked list.
-    address constant _OWNERS_LIST_SENTINEL = address(0x1);
-
-    // The mapping of identities, from identityId to IdentityDetails
     mapping(uint => IdentityDetails) public identities;
 
-    // The mapping identities owners, from identityId to OwnerList
-    mapping(address => OwnerList) public identityOwners;
-
     // Events
+    event IdentityCreated(uint indexed identityId, string name, Metadata metadata, address attestationAddr);
+    event IdentityNameUpdated(uint indexed identityId, string name);
+    event IdentityMetadataUpdated(uint indexed identityId, Metadata metadata);
```

Comment:
Does it make sense to have multiple events ? This was keeping in mind for easier indexing

Reply:
I don't see why not. I like these three.